tests/installation/welcome.pm

```diff
@@ -85,17 +92,19 @@ sub run {
     if (sle_version_at_least('15') && check_var('DISTRI', 'sle')) {
         # On s390x there will be only one product which means there is no product selection
         # On aarch64 there is only product at the moment, because HPC was not interlocked yet
-        return if (check_var('ARCH', 's390x') || check_var('ARCH', 'aarch64'));
-        assert_screen('select-product');
-        my %hotkey = (
-            sles     => 's',
-            sled     => 'u',
-            sles4sap => 'i',
-            hpc      => 'x'
-        );
-        my $product = get_required_var('SLE_PRODUCT');
-        send_key 'alt-' . $hotkey{$product};
-        assert_screen('select-product-' . $product);
+        # In upgrade mode, there is no product list shown in welcome screen
+        unless (check_var('ARCH', 's390x') || check_var('ARCH', 'aarch64') || get_var('UPGRADE')) {
```

Comment:
Aren't you changing the non-upgrade s390x/aarch64 behaviour with this one as well?

Reply:
I did. I suspected that there was an issue in old code: the matching of "languagepicked" needle was missing in regular installation tests for s390x/aarch64.
So I updated the code here and verified the change with an installation test on aarch64, see http://openqa-apac1.suse.de/tests/163#step/welcome/4

Reply:
ah, perfect :)